pkg/info/channels.go

```diff
@@ -30,37 +33,109 @@ var channelChar = &unicode.RangeTable{
 	LatinOffset: 4,
 }
 
-func GetClaimRequestChannel(service, clientID string) string {
-	return formatChannel(service, clientID, "CLAIM")
+func GetClaimRequestChannel(service, clientID string) bus.Channel {
+	return bus.Channel{
+		Legacy:  formatChannel('|', service, clientID, "CLAIM"),
+		Primary: formatClientChannel(service, clientID, "CLAIM"),
+	}
 }
 
-func GetStreamChannel(service, nodeID string) string {
-	return formatChannel(service, nodeID, "STR")
+func GetStreamChannel(service, nodeID string) bus.Channel {
+	return bus.Channel{
+		Legacy:  formatChannel('|', service, nodeID, "STR"),
+		Primary: formatClientChannel(service, nodeID, "STR"),
+	}
 }
 
-func GetResponseChannel(service, clientID string) string {
-	return formatChannel(service, clientID, "RES")
+func GetResponseChannel(service, clientID string) bus.Channel {
+	return bus.Channel{
+		Legacy:  formatChannel('|', service, clientID, "RES"),
+		Primary: formatClientChannel(service, clientID, "RES"),
+	}
 }
 
-func (i *RequestInfo) GetRPCChannel() string {
-	return formatChannel(i.Service, i.Method, i.Topic, "REQ")
+func (i *RequestInfo) GetRPCChannel() bus.Channel {
+	return bus.Channel{
+		Legacy:   formatChannel('|', i.Service, i.Method, i.Topic, "REQ"),
+		Primary:  formatServerChannel(i.Service, i.Topic, i.Queue, i.Method, "REQ"),
+		Wildcard: formatServerWildcard(i.Service, i.Topic, i.Queue),
+	}
 }
 
 func (i *RequestInfo) GetHandlerKey() string {
-	return formatChannel(i.Method, i.Topic)
+	return formatChannel('/', i.Method, i.Topic)
+}
+
+func (i *RequestInfo) GetClaimResponseChannel() bus.Channel {
+	return bus.Channel{
+		Legacy:   formatChannel('|', i.Service, i.Method, i.Topic, "RCLAIM"),
+		Primary:  formatServerChannel(i.Service, i.Topic, i.Queue, i.Method, "RCLAIM"),
+		Wildcard: formatServerWildcard(i.Service, i.Topic, i.Queue),
+	}
+}
+
+func (i *RequestInfo) GetStreamServerChannel() bus.Channel {
+	return bus.Channel{
+		Legacy:   formatChannel('|', i.Service, i.Method, i.Topic, "STR"),
+		Primary:  formatServerChannel(i.Service, i.Topic, false, i.Method, "STR"),
+		Wildcard: formatServerWildcard(i.Service, i.Topic, false),
+	}
+}
+
+var scratch = &sync.Pool{
+	New: func() any {
+		b := make([]byte, 0, 512)
+		return &b
+	},
+}
+
+func formatClientChannel(service, clientID, channel string) string {
+	p := scratch.Get().(*[]byte)
```

Comment:
is the type correct here?

Reply:
sorry. didn't read this carefully

Reply:
using a non-pointer type value in `sync.Pool` makes staticcheck upset `SA6002	Storing non-pointer values in sync.Pool allocates memory`

Reply:
Wow static check catches that. That's great.